azurerm/validators_test.go

```diff
@@ -128,3 +128,42 @@ func TestValidateIso8601Duration(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateAzureTimeZone(t *testing.T) {
+	cases := []struct {
+		Value  string
+		Errors int
+	}{
+		{
+			Value:  "UTC",
+			Errors: 0,
+		},
+		{
+			Value:  "China Standard Time",
+			Errors: 0,
+		},
+		{
+			// Valid UTC time zone
+			Value:  "UTC-11",
+			Errors: 0,
+		},
+		{
+			// Wrong casing
+			Value:  "china standard time",
+			Errors: 1,
+		},
+		{
+			// Invalid UTC time zone
+			Value:  "UTC-10",
```

Comment:
Even if azure won't accept it this we probably should not use it as a failure case because it could be confusing.

Reply:
Let me see whether Azure actually accepts it or not. If so, I will update the validation rule.

Reply:
given Azure could possibly support this as a value - it seems odd to use it as  invalid data for test purposes? can we update this to `UTC-48` which should never be a valid timezone

Reply:
Tried it with Azure, and this value is not supported. Only the UTC time offsets listed in my list is accepted by Azure.